### PR TITLE
Add gui tests

### PR DESCRIPTION
**Pull request by samuelfangjw**

### CODE REVIEW

**Comment by samuelfangjw** [comment]
Ready for review

File: src/test/java/seedu/address/ui/CompletableDeadlineCardTest.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+        // same deadline, same index -> returns true
+        CompletableDeadlineCard copy = new CompletableDeadlineCard(deadline, 0);
+        assertTrue(deadlineCard.equals(copy));
+
+        // same object -> returns true
+        assertTrue(deadlineCard.equals(deadlineCard));
+
+        // null -> returns false
+        assertFalse(deadlineCard.equals(null));
```
Comment (rubieHess) [inline src/test/java/seedu/address/ui/CompletableDeadlineCardTest.java:54]: I detect that this code is problematic. According to the [Correctness (CORRECTNESS)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#correctness-correctness), [EC: Call to equals(null) (EC_NULL_ARG)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#ec-call-to-equals-null-ec-null-arg).
This method calls equals(Object), passing a null value as the argument. According to the contract of the equals() method, this call should always return false.

File: src/test/java/seedu/address/ui/CompletableDeadlineCardTest.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    public void equals() {
+        CompletableDeadline deadline = new DeadlineBuilder().build();
+        CompletableDeadlineCard deadlineCard = new CompletableDeadlineCard(deadline, 0);
+
+        // same deadline, same index -> returns true
+        CompletableDeadlineCard copy = new CompletableDeadlineCard(deadline, 0);
+        assertTrue(deadlineCard.equals(copy));
+
+        // same object -> returns true
+        assertTrue(deadlineCard.equals(deadlineCard));
```
Comment (rubieHess) [inline src/test/java/seedu/address/ui/CompletableDeadlineCardTest.java:51]: I detect that this code is problematic. According to the [Correctness (CORRECTNESS)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#correctness-correctness), [SA: Self comparison of value with itself (SA_LOCAL_SELF_COMPARISON)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#sa-self-comparison-of-value-with-itself-sa-local-self-comparison).
This method compares a local variable with itself, and may indicate a typo or a logic error. Make sure that you are comparing the right things.

File: src/main/java/seedu/address/ui/CompletableDeadlineCard.java [hunk 3]
```diff
@@ -36,7 +36,16 @@ public CompletableDeadlineCard(CompletableDeadline deadline, int displayedIndex)
...
-        completedLabel.setText(deadline.getIsDone() ? "✔" : "");
+        completedLabel.setText(getTextToDisplay(deadline.getIsDone()));
+    }
+
+    /**
+     * Returns the text of the completed label that is to be displayed.
+     *
+     * @return {@code String} containing the text that is to be displayed.
+     */
+    public static String getTextToDisplay(boolean isDone) {
```
Comment (vevek) [inline src/main/java/seedu/address/ui/CompletableDeadlineCard.java:47]: Good idea!

File: src/main/java/seedu/address/ui/CompletableTodoCard.java [hunk 4]
```diff
@@ -36,10 +36,12 @@ public CompletableTodoCard(CompletableTodo todo, int displayedIndex) {
...
 
     /**
-     * Mark this todo as done.
+     * Returns the text of the completed label that is to be displayed.
+     *
+     * @return {@code String} containing the text that is to be displayed.
      */
-    public void markAsDone() {
-        completedLabel.setText(todo.getIsDone() ? "✔" : "");
+    public static String getTextToDisplay(boolean isDone) {
```
Comment (vevek) [inline src/main/java/seedu/address/ui/CompletableTodoCard.java:43]: So for now there is not mechanism to tick the todo off in via the GUI right? It'll just update the GUI accordingly?

File: src/test/java/guitests/guihandles/CompletableDeadlineCardHandle.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+    public String getCompleted() {
+        return completedLabel.getText();
+    }
+
+    /**
+     * Returns true if this handle contains a {@code CompletableDeadline}.
+     */
+    public boolean equals(CompletableDeadline deadline) {
+        return getDescription().equals(deadline.getDescription())
```
Comment (vevek) [inline src/test/java/guitests/guihandles/CompletableDeadlineCardHandle.java:57]: Was id left out intentionally?

File: src/test/java/guitests/guihandles/CompletableTodoCardHandle.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+    public String getCompleted() {
+        return completedLabel.getText();
+    }
+
+    /**
+     * Returns true if this handle contains a {@code CompletableTodo}.
+     */
+    public boolean equals(CompletableTodo todo) {
+        return getDescription().equals(todo.getDescription())
```
Comment (vevek) [inline src/test/java/guitests/guihandles/CompletableTodoCardHandle.java:48]: Is the ID here intentionally left out?

File: src/test/java/guitests/guihandles/EventCardHandle.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+    public String getDate() {
+        return dateLabel.getText();
+    }
+
+    /**
+     * Returns true if this handle contains an {@code Event}.
+     */
+    public boolean equals(Event event) {
+        return getDescription().equals(event.getDescription())
```
Comment (vevek) [inline src/test/java/guitests/guihandles/EventCardHandle.java:56]: id here too

File: src/test/java/seedu/address/ui/CompletableDeadlineCardTest.java [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+        // null -> returns false
+        assertFalse(deadlineCard.equals(null));
+
+        // different types -> returns false
+        assertFalse(deadlineCard.equals(0));
+
+        // different deadline, same index -> returns false
```
Comment (vevek) [inline src/test/java/seedu/address/ui/CompletableDeadlineCardTest.java:59]: Reading this, it looks like ID should be checked too. Hmm

File: src/test/java/seedu/address/ui/CompletableDeadlineCardTest.java [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+        CompletableDeadline differentDeadline = new DeadlineBuilder().withDescription("differentDescription").build();
+        assertFalse(deadlineCard.equals(new CompletableDeadlineCard(differentDeadline, 0)));
+
+        // same deadline, different index -> returns false
+        assertFalse(deadlineCard.equals(new CompletableDeadlineCard(deadline, 1)));
```
Comment (vevek) [inline src/test/java/seedu/address/ui/CompletableDeadlineCardTest.java:64]: here too.

**Review by vevek — APPROVED** [review]
LGTM, but see the comments first

File: src/test/java/guitests/guihandles/CompletableDeadlineCardHandle.java (hunk 5 above)
Comment (samuelfangjw) [reply src/test/java/guitests/guihandles/CompletableDeadlineCardHandle.java:57]: ID was left out intentionally, copied this method from another handle object. Purpose of this is to compare this handler object to another completable deadline, which doesn’t have an id. I’ll look into changing the naming for these methods to clear things up.

Comment (vevek) [reply src/test/java/guitests/guihandles/CompletableDeadlineCardHandle.java:57]: Got it man, thanks

File: src/main/java/seedu/address/ui/CompletableTodoCard.java (hunk 4 above)
Comment (samuelfangjw) [reply src/main/java/seedu/address/ui/CompletableTodoCard.java:43]: yes, only can tick off using mark as done.

**Pull request merged**